src/services/blocks/BlocksService.ts

```diff
@@ -251,21 +265,35 @@ export class BlocksService extends AbstractService {
 							weightInfoClass
 						] as WeightPerClass).baseExtrinsic;
 				} else {
-					// We are querying a runtime that matches the decorated metadata in the api
+					// We are querying a runtime that matches the metadata in the api
 					extrinsicBaseWeight =
 						(api.consts.system?.extrinsicBaseWeight as AbstractInt) ||
 						(api.consts.system.blockWeights.perClass[
 							weightInfoClass
 						] as WeightPerClass).baseExtrinsic;
 				}
 
+				if (extrinsicBaseWeight === undefined) {
+					extrinsics[idx].info = {
+						error: 'Error occured while finding the extrinsicBaseWeight',
+					};
+					continue;
+				}
+
+				let calcFeeExBaseWeight;
+				if (typeof extrinsicBaseWeight !== 'number') {
+					calcFeeExBaseWeight = extrinsicBaseWeight.toBigInt();
+				} else {
+					calcFeeExBaseWeight = BigInt(extrinsicBaseWeight);
+				}
+
 				const len = block.extrinsics[idx].encodedLength;
 				const weight = weightInfo.weight;
 
 				const partialFee = calcFee.calc_fee(
 					BigInt(weight.toString()),
 					len,
-					extrinsicBaseWeight.toBigInt()
+					calcFeeExBaseWeight
 				);
 
 				extrinsics[idx].info = api.createType('RuntimeDispatchInfo', {
```

Comment:
Commenting here because I can't comment on line 304 because GitHub. It used to be that try-catch blocks were fairly expensive performance and memory wise. It's not immediately obvious to me what it buys us here as opposed to having an `all_good` boolean and, if it's false, print all `extrinsics` with an error, in the end?

Or maybe V8 is good enough these days that it doesn't matter?

Reply:
@TarikGul ^^^ didn't pan out? Or am I wrong?

Reply:
So `try-catch` block's themselves is actually pretty fast these days if an exception is not being thrown and as far as I know, from the research I did it has almost zero penalty (This is excluding throwing exceptions). In this case I don't think its absolutely necessary to remove the `try-catch` because of all the checks before calcFee is called which significantly removes the chance of an exception being thrown. That being said though if we start seeing exceptions thrown in the `try-catch` it will reveal deeper issues with `calc_fee`, or interdependent logic, which in retrospect should be caught beforehand if we have the necessary testing.